ru/lessons/basics/control-structures.md

```diff
@@ -127,3 +128,53 @@ iex> cond do
 ...> end
 "Catch all"
 ```
+
+## `with`
+
+Специальная форма `with` может пригодиться в ситуациях, когда сложно использовать оператор потока, либо когда нужен вложенный вызов `case`. `with` состоит из ключевого слова, генераторов и, в конце, выражения.
+
+Мы ещё обсудим генераторы в уроке о списковых включениях, но сейчас нам достаточно знать лишь то, что они используют сопоставление с образцом для сравнения правой части `<-` с левой.
```

Comment:
I think that adding "list comprehension" term in brackets near the first term appearance may be good practice. But it's up to you.